spec/integration/outputs/index_spec.rb

```diff
@@ -0,0 +1,129 @@
+# encoding: utf-8
+require "logstash/devutils/rspec/spec_helper"
+require "logstash/outputs/elastic_app_search"
+require "logstash/codecs/plain"
+require "logstash/event"
+require "json"
+
+describe "indexing against running AppSearch", :integration => true do
+
+  let(:engine_name) {
+    (0...10).map { ('a'..'z').to_a[rand(26)] }.join
+  }
+
+  let(:config) {
+    {
+      "api_key" => ENV['APPSEARCH_PRIVATE_KEY'],
+      "engine" => engine_name,
+      "url" => "http://appsearch:3002"
+    }
+  }
+
+  subject { LogStash::Outputs::ElasticAppSearch.new(config) }
+
+  before(:each) do
+    create_engine(engine_name, "http://appsearch:3002", ENV['APPSEARCH_PRIVATE_KEY'])
+  end
+
+  private
+  def create_engine(engine_name, host, api_key)
+    url = host + "/api/as/v1/engines"
+    resp = Faraday.post(url, "{\"name\": \"#{engine_name}\"}",
+      "Content-Type" => "application/json",
+      "Authorization" => "Bearer " + api_key)
+    expect(resp.status).to eq(200)
+  end
+
+  describe "search and private keys are configured" do
+    let(:api_key_settings) {
+      {
+        :private => ENV['APPSEARCH_PRIVATE_KEY'],
+        :search => ENV['APPSEARCH_SEARCH_KEY']
+      }
+    }
```

Comment:
Nitpick: multiline `do`...`end`

```suggestion
    let(:api_key_settings) do
      {
        :private => ENV['APPSEARCH_PRIVATE_KEY'],
        :search => ENV['APPSEARCH_SEARCH_KEY']
      }
    end
```